contracts/deposit-contracts/contract/src/main.rs

```diff
@@ -0,0 +1,218 @@
+#![no_std]
+#![no_main]
+extern crate alloc;
+use alloc::{collections::BTreeMap, string::ToString, vec, vec::Vec};
+use casper_contract::{
+    contract_api::{runtime, storage, system},
+    unwrap_or_revert::UnwrapOrRevert,
+};
+use casper_types::{
+    bytesrepr::ToBytes, contracts::NamedKeys, runtime_args, AccessRights, ApiError, CLValue,
+    EntryPoints, Key, RuntimeArgs, URef, U512,
+};
+mod constants;
+use constants::{
+    ADMIN_LIST, KAIROS_DEPOSIT_CONTRACT, KAIROS_DEPOSIT_CONTRACT_NAME,
+    KAIROS_DEPOSIT_CONTRACT_PACKAGE, KAIROS_DEPOSIT_EVENT_DICT, KAIROS_DEPOSIT_PURSE,
+    KAIROS_LAST_PROCESSED_DEPOSIT_COUNTER, KAIROS_MOST_RECENT_DEPOSIT_COUNTER, RUNTIME_ARG_AMOUNT,
+    RUNTIME_ARG_DEST_PURSE, RUNTIME_ARG_TEMP_PURSE, SECURITY_BADGES,
+};
+mod detail;
+use detail::{get_immediate_caller, get_optional_named_arg_with_user_errors};
+mod error;
+use error::DepositError;
+mod security;
+use security::{access_control_check, SecurityBadge};
+mod entry_points;
+
+use contract_types::Deposit;
+
+// This entry point is called once when the contract is installed
+// and sets up the security badges with the installer as an admin or the
+// optional list of admins that was passed to the installation session as a runtime argument.
+// The contract purse will be created in contract context so that it is "owned" by the contract
+// instead of the installing account.
+#[no_mangle]
+pub extern "C" fn init() {
```

Comment:
The installer will by default always be an admin. If you pass a list that contains the installer it will be added only once / overriden. Should you not supply a list then there is only 1 admin. 

If you want to remove the admin from the list you will have to do that after the contract was installed. 
The way this is done is negotiable but a default admin is required because otherwise noone can udpate the list. The default admin could later be another contract or a multisig wallet, all of this will depend on our implementaiton.

Since I have removed all Withdrawal functionality from the contract as per Andrzej's request, we currently only need to restrict access to the "increase counter" function for the sliding window technique that was discussed during our design-sessions. The admin account will be responsible for updating the on-chain counter every time that a deposit was included in a batch / processed.